lua/pretty-ts-errors/diagnostics.lua

```diff
@@ -138,8 +147,14 @@ function M.show_formatted_error(opts)
 				width = math.min(width, config.get().float_opts.max_width)
 				local height = math.min(#lines, config.get().float_opts.max_height)
 
+				if window == nil then
+					-- If the window is nil, that means we're in lazy window creation mode,
+					-- so initialise it here.
+					window = init_window(main_buf, floating_buf, win_opts)
+				end
```

Comment:
Should maybe go further and only render the window once all diagnostics have been resolved, not just the first once...